dapp/src/services/price.service.js

```diff
@@ -0,0 +1,12 @@
+export default class PriceService {
+  async fetchPrice() {
+    const endpoint = `${process.env.COINGECKO_API}/simple/price?ids=origin-protocol%2Corigin-dollar-governance&vs_currencies=usd`
```

Comment:
Question: Does CoinGecko's free tier API have any sort of rate-limit? If yes, could it pose a problem during SSR?